aiidalab/app.py

```diff
@@ -228,12 +235,29 @@ def find_matching_releases(self, specifier, prereleases=None):
         matching_releases.sort(key=parse, reverse=True)
         return matching_releases
 
+    @staticmethod
+    def _strict_dependencies_met(requirements, python_bin):
+        """Check whether the given requirements are compatible with the core dependencies of a package."""
```

Comment:
```suggestion
        """Check whether the given requirements are compatible with the core dependencies of a package."""
            if requirements is None or len(requirements) == 0:
                return True 
```

Reply:
This also covered by the default value empty list `[]` returned no dependencies defined in the file.

Reply:
But in this case, I realised that using the python type hint will prevent lots of cases where these issues may happend. I'll open an issue to bring the mypy into these packages, at least for `aiidalab` and `aiidalab-home` first.